examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -41,6 +44,203 @@ using namespace chip::DeviceLayer;
+// using expected commands like CTRL-C to quit the application. (see issue #17845)
+// We should stop using signals for those faults, and move to a different notification
+// means, like a pipe. (see issue #19114)
+#if !defined(ENABLE_CHIP_SHELL)
+void OnRebootSignalHandler(int signum)
+{
+    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+
+    // The BootReason attribute SHALL indicate the reason for the Node’s most recent boot, the real usecase
+    // for this attribute is embedded system. In Linux simulation, we use different signals to tell the current
+    // running process to terminate with different reasons.
+    BootReasonType bootReason = BootReasonType::kUnspecified;
+    switch (signum)
+    {
+    case SIGVTALRM:
+        bootReason = BootReasonType::kPowerOnReboot;
+        break;
+    case SIGALRM:
+        bootReason = BootReasonType::kBrownOutReset;
+        break;
+    case SIGILL:
+        bootReason = BootReasonType::kSoftwareWatchdogReset;
+        break;
+    case SIGTRAP:
+        bootReason = BootReasonType::kHardwareWatchdogReset;
+        break;
+    case SIGIO:
+        bootReason = BootReasonType::kSoftwareUpdateCompleted;
+        break;
+    case SIGINT:
+        bootReason = BootReasonType::kSoftwareReset;
+        break;
+    default:
+        IgnoreUnusedVariable(bootReason);
+        ChipLogError(NotSpecified, "Unhandled signal: Should never happens");
+        chipDie();
+        break;
+    }
+
+    Server::GetInstance().DispatchShutDownAndStopEventLoop();
+}
+
+void OnSoftwareFaultSignalHandler(int signum)
+{
+    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+
+    VerifyOrDie(signum == SIGUSR1);
+    PlatformMgr().ScheduleWork(HandleSoftwareFaultEvent);
+}
+
+void OnGeneralFaultSignalHandler(int signum)
+{
+    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+
+    uint32_t eventId;
+    switch (signum)
+    {
+    case SIGUSR2:
+        eventId = Clusters::GeneralDiagnostics::Events::HardwareFaultChange::Id;
+        break;
+    case SIGHUP:
+        eventId = Clusters::GeneralDiagnostics::Events::RadioFaultChange::Id;
+        break;
+    case SIGTTIN:
+        eventId = Clusters::GeneralDiagnostics::Events::NetworkFaultChange::Id;
+        break;
+    default:
+        ChipLogError(NotSpecified, "Unhandled signal: Should never happens");
+        chipDie();
+        break;
+    }
+
+    PlatformMgr().ScheduleWork(HandleGeneralFaultEvent, static_cast<intptr_t>(eventId));
+}
+
+void SetupSignalHandlers()
+{
+    // sigaction is not used here because Tsan interceptors seems to
+    // never dispatch the signals on darwin.
+    signal(SIGALRM, OnRebootSignalHandler);
+    signal(SIGVTALRM, OnRebootSignalHandler);
+    signal(SIGILL, OnRebootSignalHandler);
+    signal(SIGTRAP, OnRebootSignalHandler);
+    signal(SIGTERM, OnRebootSignalHandler);
+    signal(SIGIO, OnRebootSignalHandler);
+    signal(SIGINT, OnRebootSignalHandler);
```

Comment:
Please do not override SIGINT

Reply:
replace with SIGSTP temperately, and this whole mechanism will be replaced by pipe